collect_app/src/main/java/org/odk/collect/android/database/entities/DatabaseEntitiesRepository.kt

```diff
@@ -213,12 +214,20 @@ class DatabaseEntitiesRepository(context: Context, dbPath: String) : EntitiesRep
             return emptyList()
         }
 
-        return queryWithAttachedRowId(
-            list,
-            selectionColumn = property,
-            selectionArg = value
-        ).foldAndClose {
-            mapCursorRowToEntity(list, it, it.getInt(ROW_ID))
+        return if (databaseConnection.readableDatabase.doesColumnExist(list, property)) {
+            queryWithAttachedRowId(
+                list,
+                selectionColumn = property,
+                selectionArg = value
+            ).foldAndClose {
+                mapCursorRowToEntity(list, it, it.getInt(ROW_ID))
+            }
+        } else if (value == "") {
```

Comment:
Can properties have blank values like `" "`? Is it sanitized somewhere? I'm wondering if here we should use `value.isBlank()` instead.

Reply:
I'm realizing the behaviour here is wrong. The test describes it correctly ("getByAllByProperty returns entities **without** property when searching for empty string"), but doesn't actually test the right thing. I'll fix it.

Reply:
> I see but isn't it somehow possible that it doesn't exist and the value here is not "" but " "?

Yes, but in that case we'd return an empty list because we'd be searching for nodes with a non-existent property equal to "something" rather than "nothing". JavaRosa treats non-existent nodes as having the value `""`.